From d63354c3aeff63bc09dfe120b4a85ecf96549215 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 09 Feb 2020 08:50:02 +0200
Subject: [PATCH] [SW-8229] goya2: split perstn reset to target specific code

On onix setting the PERSTN reset will reset the ARC. So we split
this function to target specific files.

Change-Id: I67f42f9ebbb5b16a5cbe846f64a4b29b9d44d031
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/Makefile b/Makefile
index 7729326..330c3e5 100644
--- a/Makefile
+++ b/Makefile
@@ -40,10 +40,12 @@
 ARCBOOT_COMMON_SRCS_C := $(shell ls common/*.c)
 
 ARCBOOT_RAM_SRCS_C := app/arcboot_ram.c
+ARCBOOT_RAM_SRCS_C += $(HL_DEV)/$(HL_TARGET).c
 ARCBOOT_RAM_SRCS_S := app/crt0_ram.S
 ARCBOOT_RAM_OBJECTS := $(ARCBOOT_COMMON_SRCS_C:.c=.o) $(ARCBOOT_RAM_SRCS_C:.c=.o) $(ARCBOOT_RAM_SRCS_S:.S=.o)
 
 ARCBOOT_ROM_SRCS_C := app/arcboot_rom.c
+ARCBOOT_ROM_SRCS_C += $(HL_DEV)/$(HL_TARGET).c
 ARCBOOT_ROM_SRCS_S := app/crt0_rom.S
 ARCBOOT_ROM_OBJECTS := $(ARCBOOT_COMMON_SRCS_C:.c=.o) $(ARCBOOT_ROM_SRCS_C:.c=.o) $(ARCBOOT_ROM_SRCS_S:.S=.o)
 
diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 6cce9b4..a4a740a 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -30,12 +30,8 @@
 	/* make sure spi controller doesn't swap bytes */
 	spi_flash_set_byte_swap(DISABLE_BYTE_SWAP);
 
-	/* on warm reset change default to reset ARC and PSOC */
 	/* TODO: ask raviv why in fpga we get a reset on start */
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
-	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, reg);
+	hl_arcboot_rom_prstn_reset_config();
 
 	/* read from cold reset flops the recovery and validation flag */
 	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
new file mode 100644
index 0000000..a0b1b12
--- /dev/null
+++ b/goya2_arc/asic.c
@@ -0,0 +1,19 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include "arcboot.h"
+
+void hl_arcboot_rom_prstn_reset_config(void)
+{
+	uint32_t reg;
+
+	/* on warm reset change default to reset ARC and PSOC */
+	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
+	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, reg);
+}
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
new file mode 100644
index 0000000..ecbdd2d
--- /dev/null
+++ b/goya2_arc/onix.c
@@ -0,0 +1,10 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include "arcboot.h"
+
+void hl_arcboot_rom_prstn_reset_config(void) {}
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
new file mode 100644
index 0000000..a0b1b12
--- /dev/null
+++ b/goya2_arc/pldm.c
@@ -0,0 +1,19 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include "arcboot.h"
+
+void hl_arcboot_rom_prstn_reset_config(void)
+{
+	uint32_t reg;
+
+	/* on warm reset change default to reset ARC and PSOC */
+	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
+	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, reg);
+}
diff --git a/include/arcboot.h b/include/arcboot.h
index 71cba7b..87c0efc 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -57,6 +57,7 @@
 };
 
 extern struct arcboot_global_s arcboot_rom_global;
+void hl_arcboot_rom_prstn_reset_config(void);
 
 enum security_location {
 	ASIC_EFUSE,
